frontend/src/scenes/surveys/Survey.tsx

```diff
@@ -51,6 +50,19 @@ export function SurveyForm({ id }: { id: string }): JSX.Element {
     const { survey, surveyLoading, isEditingSurvey, targetingFlagFilters } = useValues(surveyLogic)
     const { loadSurvey, editingSurvey } = useActions(surveyLogic)
 
+    const handleCancelClick = (): void => {
+        if (isEditingSurvey) {
+            editingSurvey(false)
+            if (id === 'new') {
+                router.actions.push(urls.surveys())
+            } else {
+                loadSurvey()
+            }
+        } else {
+            router.actions.push(urls.surveys())
+        }
+    }
```

Comment:
This fixes the `Cancel` button issue I called out here: https://posthog.slack.com/archives/C034XD440RK/p1716327059331599

Reply:
small readability nit - I think this is equivalent, but let me know if I'm missing something!

```suggestion
    const handleCancelClick = (): void => {
        editingSurvey(false)
        if (id === 'new') {
            router.actions.push(urls.surveys())
        } else {
            loadSurvey()
        }
    }
```